### ISSUE TITLE
Impossible to use double role assuming profile authentication

### ISSUE THREAD

**Issue by mikaelrandy** [terraform-providers/terraform-provider-aws#5251]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

```
$ terraform -v
Terraform v0.11.7
+ provider.aws v1.27.0
```

### Terraform Configuration Files
`~/.aws/credentials`
```
[root]
aws_access_key_id = ***
aws_secret_access_key = ***
region = ***
```

`~/.aws/config`
```
[profile RoleForTech]
source_profile = root
role_arn = arn:aws:iam::[ROOT ACCOUNT]:role/RoleForTech

[profile development]
source_profile = RoleForTech
role_arn = arn:aws:iam::[DEV ACCOUNT]:role/tech
```

`teams.tf`
```
terraform {
  backend "s3" {
    bucket         = "foobar"
    dynamodb_table = "foobar"
    encrypt        = true
    kms_key_id     = "foobar"
    region         = "eu-west-3"
    profile        = "development"
  }
}

```

### Debug Output

https://gist.github.com/mikaelrandy/e41b93311c679b45a1496766e16f0ecb

### Expected Behavior

Terraform successfully initiate my repository, using my `development` AWS profile.

NB : executing AWS CLI command with this profile works perfectly.

### Actual Behavior

AWS authentication error : 
```
Error configuring the backend "s3": Error creating AWS session: SharedConfigAssumeRoleError: failed to load assume role for arn:aws:iam::311275790335:role/tech, source profile has no shared credentials
```

### Steps to Reproduce

1. `terraform init -backend-config=key=service-6play-ad-interceptor-api.tfstate`

### References

It's seem it's a bug in AWS Go SDK.
I opened an issue : https://github.com/aws/aws-sdk-go/issues/2063

This SDK will have to be updated here once the issue will be fixed.

**Comment by henrydobson** [comment]
@paddycarver are there any plans to look at this issue currently?

**Comment by AdeOpe** [comment]
any ideas on how to workaroud this?

**Comment by mikaelrandy** [comment]
@AdeOpe in my use case, I change my authentication architecture to avoid double assume role.
Sorry :/

**Comment by AdeOpe** [comment]
@mikaelrandy thanks for the reply. i had to do the same also.

**Comment by YakDriver** [comment]
I believe this is fixed in hashicorp/aws-sdk-go-base#5. If someone can test that would be 💯 .

**Comment by aeschright** [comment]
I confirmed that this will be fixed with aws-sdk-go-base@v0.4.0 in #10379. Adding it to the next release milestone.

**Comment by bflad** [comment]
Closing as #10379 was merged previously and v2.32.0 has been released. 👍
